app/helpers/results_helper.rb

```diff
@@ -13,21 +13,29 @@ def result_groups(session)
   end
 
   def filter_questions_by_session(group_key, session)
-    I18n.t("results_link.#{group_key}").each_with_object([]) do |question, array|
+    group = I18n.t("results_link.#{group_key}").dup
+    group.each_with_object([]) do |question, array|
       if question[1][:show_options].include?(session[question[0]])
-        array << I18n.t("results_link.#{group_key}.#{question[0]}")
+        array << filter_results_by_country(I18n.t("results_link.#{group_key}.#{question[0]}").dup)
```

Comment:
Is the duplication (`.dup`) of the hash necessary here and on line 16?  It seems to make no difference when I remove them and run the application locally.